controller/src/main/java/io/pravega/controller/eventProcessor/impl/SerializedRequestHandler.java

```diff
@@ -82,8 +86,35 @@
      */
     private void run(String key, ConcurrentLinkedQueue<Work> workQueue) {
         Work work = workQueue.poll();
-        FutureHelpers.completeAfter(() -> processEvent(work.getEvent()), work.getResult());
-        work.getResult().whenComplete((r, e) -> {
+        processEvent(work.getEvent()).whenComplete((r, e) -> {
+            if (e != null && toPostpone(work.getEvent(), work.getPickupTime(), e)) {
+                handleWorkPostpone(key, workQueue, work);
+            } else {
+                if (e != null) {
+                    work.getResult().completeExceptionally(e);
+                } else {
+                    work.getResult().complete(r);
+                }
+
+                handleWorkComplete(key, workQueue, work);
+            }
+        });
+    }
+
+    private void handleWorkPostpone(String key, ConcurrentLinkedQueue<Work> workQueue, Work work) {
+        // if the request handler decides to postpone the processing,
+        // put the work at the back of the queue to be picked again.
+        // Note: we have not completed the work's result future here.
+        // Since there is at least one event in the queue (we just added) so we will call run again.
```

Comment:
Minor: If you do another round of changes, please break this line to match the end of the others.